### PR TITLE
Fix TLS (`client_encryption_options`) configuration for CQL

### LINKED ISSUE #1001: Cql client encryption is not working
Hi,

I am trying to use stargate 1.0.25 in front of Cassandra 3.11 with enabling ssl for both server and client. Server ssl is working.
But client encryption is not working. 

```
env JAVA_OPTS='-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005 -Dstargate.unsafe.cassandra_config_path=/stargate/stargate/cass_configs/cassandra.yaml' \
./starctl --cluster-name cass \
--cluster-seed 10.10.10.21 \
--cluster-version 3.11 \
--listen 10.0.0.161 \
--dc cass \
--rack 3a \
--enable-auth

```

In cassandra.yaml, I have

```
client_encryption_options:
    enabled: true
    optional: true
    require_client_auth: true

```
In log, I can see:

`INFO  [main] 2021-06-08 15:11:05,959 Server.java:157 - Starting listening for CQL clients on /0.0.0.0:9042 (unencrypted)...`
    
Is there any other configuration that I should use to enable client ssl for cql?

Thanks.

### PR DESCRIPTION
**Pull request by mpenick**
**What this PR does**:

CQL TLS configuration is now loaded from a yaml  file specified using the system property `stargate.cql.config_path`. This very similar to `cassandra.yaml`, but it only contains the options that are specific to the CQL transport layer.

This also include a couple bug fixes:
* Removes usage of `ClientWarn`, this is handled by the persistence layer.
* Remove usage of `DatabaseDescriptor` which is not properly configured.

Substantial portions of the following files have been copied from C* 4.0.3:
* Config.java
* EncryptionOptions.java
* YamlConfigurationLoader.java
* SSLFactory.java

**Which issue(s) this PR fixes**:
Fixes #1001

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] Add an example `cql.yaml`  file to resources
- [x] Add tests for client certificates
- [x] Fix licences on `.java` files (either Stargate or Apache Cassandra)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
One thing that may or may not be relevant: instead of using SnakeYAML directly (btw do we not need a dependency from `pom.xml` to it? Or is it assumed we get it transitively from Cassandra), there's also Jackson YAML module:

https://github.com/FasterXML/jackson-dataformats-text/tree/2.14/yaml

which let's you use Jackson API to bind to POJOs, or `Map`s or whatever using YAML-backed `ObjectMapper`.
That could simplify things a bit. It does use SnakeYAML for actual decoding fwtw.

File: cql/src/main/java/io/stargate/cql/impl/CqlImpl.java [hunk 1]
```diff
@@ -85,7 +85,6 @@ public void start() {
             .withEventLoopGroup(workerGroup)
             .withHost(nativeAddr);
 
-    TransportDescriptor.getNativeProtocolEncryptionOptions().applyConfig();
     if (!TransportDescriptor.getNativeProtocolEncryptionOptions().isEnabled()) {
```
[the remark this answers is not in the document]
Comment (mpenick) [reply cql/src/main/java/io/stargate/cql/impl/CqlImpl.java:88]: Thanks that's a good catch. Thanks

[the remark this answers is not in the document]
Comment (mpenick) [reply cql/src/main/java/io/stargate/cql/impl/CqlImpl.java:88]: Fixed.

**Comment by mpenick** [comment]
> One thing that may or may not be relevant: instead of using SnakeYAML directly (btw do we not need a dependency from `pom.xml` to it? Or is it assumed we get it transitively from Cassandra), there's also Jackson YAML module:
> 
> https://github.com/FasterXML/jackson-dataformats-text/tree/2.14/yaml
> 
> which let's you use Jackson API to bind to POJOs, or `Map`s or whatever using YAML-backed `ObjectMapper`. That could simplify things a bit. It does use SnakeYAML for actual decoding fwtw.

Updated in this commit: https://github.com/stargate/stargate/pull/1992/commits/de500843beea07c377eaa381a46a6af01279d004

File: cql/src/main/java/io/stargate/cql/CqlActivator.java [hunk 2]
```diff
@@ -15,21 +15,30 @@
...
-import java.net.UnknownHostException;
 import java.util.Arrays;
 import java.util.List;
 import javax.annotation.Nullable;
-import org.apache.cassandra.config.Config;
+import org.apache.cassandra.stargate.config.Config;
 
 public class CqlActivator extends BaseActivator {
+  public static ObjectMapper mapper =
+      new ObjectMapper(new YAMLFactory())
```
Comment (tatu-at-datastax) [inline cql/src/main/java/io/stargate/cql/CqlActivator.java:39]: not a big deal but there's also `YAMLMapper` that constructs `YAMLFactory` (no need to change if no other changes made), which is a sub-class of `ObjectMapper`

File: cql/src/main/resources/cql.yaml [hunk 3]
```diff
@@ -0,0 +1,126 @@
+
+# CQL transport config YAML
```
Comment (olim7t) [inline cql/src/main/resources/cql.yaml:2]: It looks like this file is not referenced anywhere. Is it intended as an example?

**Review by olim7t — APPROVED** [review]

File: cql/src/main/resources/cql.yaml (hunk 3 above)
Comment (mpenick) [reply cql/src/main/resources/cql.yaml:2]: Yes, it's an an example file.

**Pull request merged**